### PR TITLE
solution

### PR DESCRIPTION
**Pull request by abcdesign1**
[DEMO LINK](https://vadym4che.github.io/react_tabs-with-router/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,12 +1,78 @@
...
+const getTabById = (tabsArr: Tab[], idFindBy: string) => {
+  return tabsArr.find(tab => tab.id === idFindBy) || null;
+};
+
+const Tabs: FC<Props> = ({ tabsArray, tabId }) => {
+  const [selectedTab, setSelectedTab] = useState<Tab | null>(null);
+
+  useEffect(() => {
+    setSelectedTab(getTabById(tabsArray, tabId));
+  }, [tabId]);
```
Comment (andriibezkorovainyi) [inline src/App.tsx:40]: You can save selectedTab just in variable without useState:
```suggestion
  const selectedTab = getTabById(tabsArray, tabId);
  const isTabSelected = (tab: Tab) => tab.id === tabId;
```

File: src/App.tsx [hunk 2]
```diff
@@ -1,12 +1,78 @@
...
+    <>
+      <div className="tabs is-boxed">
+        <ul>
+          {tabsArray.map(tab => {
+            return (
+              <li
+                data-cy="Tab"
+                className={`${tab.id === tabId ? 'is-active' : ''}`}
+              >
+                <Link to={`/tabs/${tab.id}`}>{tab.title}</Link>
```
Comment (andriibezkorovainyi) [inline src/App.tsx:52]: If you want to make your component more universal, don't hardcode the address
```suggestion
                <Link to={`.../${tab.id}`}>{tab.title}</Link>
```

File: src/App.tsx [hunk 3]
```diff
@@ -17,35 +83,59 @@ export const App = () => (
     >
       <div className="container">
         <div className="navbar-brand">
-          <a href="/" className="navbar-item is-active">Home</a>
-          <a href="/tabs" className="navbar-item">Tabs</a>
+          <NavLink
+            className={({ isActive }) => `navbar-item ${isActive ? 'is-active' : ''}`}
```
Comment (andriibezkorovainyi) [inline src/App.tsx:87]:
```suggestion
            className={({ isActive }) => `isTabSelected(tab)`}
```

File: src/App.tsx [hunk 4]
```diff
@@ -17,35 +83,59 @@ export const App = () => (
...
+            to="/"
+          >
+            Home
+          </NavLink>
+
+          <NavLink
+            className={({ isActive }) => `navbar-item ${isActive ? 'is-active' : ''}`}
```
Comment (andriibezkorovainyi) [inline src/App.tsx:94]: same

**Review by andriibezkorovainyi — APPROVED** [review]
GJ, I've found some minor changes i would change. And create separate files for your components - it's difficult to read.

File: src/App.tsx (hunk 1 above)
Comment (OleksiiBlezniuk) [inline src/App.tsx:40]: Why do you need useEffect here?

Comment (OleksiiBlezniuk) [reply src/App.tsx:40]: Most likely if tabsArray will be changed too - you'll need to find object again

File: src/App.tsx [hunk 5]
```diff
@@ -1,12 +1,78 @@
...
+interface Props {
+  tabsArray: Tab[];
+  tabId: string;
+}
+
+const getTabById = (tabsArr: Tab[], idFindBy: string) => {
+  return tabsArr.find(tab => tab.id === idFindBy) || null;
+};
+
+const Tabs: FC<Props> = ({ tabsArray, tabId }) => {
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:35]: Each component should be in separate file

File: src/App.tsx (hunk 2 above)
Comment (OleksiiBlezniuk) [reply src/App.tsx:52]: Agree, fix it

File: src/App.tsx [hunk 6]
```diff
@@ -1,12 +1,78 @@
...
+              >
+                <Link to={`/tabs/${tab.id}`}>{tab.title}</Link>
+              </li>
+            );
+          })}
+        </ul>
+      </div>
+
+      <div className="block" data-cy="TabContent">
+        {selectedTab === null ? 'Please select a tab' : selectedTab.content}
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:60]:
```suggestion
        {!selectedTab ? 'Please select a tab' : selectedTab.content}
```

File: src/App.tsx [hunk 7]
```diff
@@ -1,12 +1,78 @@
...
+      </div>
+    </>
+  );
+};
+
+const TabsPage = () => {
+  const { tabId } = useParams();
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:67]: Better to create type for params and pass it as generic here

File: src/App.tsx (hunk 3 above)
Comment (OleksiiBlezniuk) [inline src/App.tsx:87]: use classnames for this purpose

File: src/App.tsx [hunk 8]
```diff
@@ -17,35 +83,59 @@ export const App = () => (
...
+          >
+            Home
+          </NavLink>
+
+          <NavLink
+            className={({ isActive }) => `navbar-item ${isActive ? 'is-active' : ''}`}
+            to="/tabs"
+          >
+            Tabs
+          </NavLink>
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:98]: Move NavLink to separate component to avoid duplicating className prop

File: src/App.tsx [hunk 9]
```diff
@@ -17,35 +83,59 @@ export const App = () => (
...
+          <Route
+            path="/home"
+            element={
+              <Navigate to="/" replace />
+            }
+          />
+
+          <Route
+            path="/tabs/"
+          >
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:120]:
```suggestion
          <Route path="/tabs/">
```

File: src/App.tsx [hunk 10]
```diff
@@ -17,35 +83,59 @@ export const App = () => (
...
+            <Route
+              index
+              element={<TabsPage />}
+            />
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:124]:
```suggestion
            <Route index element={<TabsPage />} />
```

**Review by OleksiiBlezniuk — CHANGES_REQUESTED** [review]

**Pull request closed without merging**